src/backend/core/api/serializers.py

```diff
@@ -226,6 +229,23 @@ def validate_file(self, file):
                 f"File type '{mime_type:s}' is not allowed. Allowed types are: {mime_types:s}"
             )
 
+        # Compress the image if it’s a valid image type
+        if mime_type and mime_type.startswith("image/"):
```

Comment:
I'm not sure this test is enough, because I guess it will fail for `image/svg+xml`. `image = Image.open(file)` will probably raise an error + I'm not sure we can "compress" svg files.

Maybe we should use a 
``` python
try:
    image = Image.open(file)
except IOError:
    # not an image
else:
    # compress
```